src/utils.js

```diff
@@ -71,3 +71,15 @@ export function findUnit({ subworkflowData, index, type }) {
     if (unit.type !== type) throw new Error("findUnit() error: unit type does not match!");
     return unit;
 }
+
+// Function to sort array based on the order given in a separate array
+export function sortArrayByOrder(arr, order) {
```

Comment:
This should be in `@mat3ra/utils` array.ts - there might be already a function for it